src/panels/lovelace/cards/hui-fan-card.ts

```diff
@@ -0,0 +1,357 @@
+
+        <div class="content">
+          <div id="controls">
+            <div id="slider">
+              <round-slider
+                min="0"
+                max=${max_speed}
+                .value=${speed}
+                .disabled=${UNAVAILABLE_STATES.includes(stateObj.state)}
+                @value-changing=${this._dragEvent}
+                @value-changed=${this._setSpeed}
+                style=${styleMap({
+                  visibility: "visible",
+                })}
+              ></round-slider>
+              <ha-icon-button
+                class="fan-button ${classMap({
+                  "slider-center": true,
+                  "state-on": stateObj.state === "on",
+                  "state-unavailable": stateObj.state === UNAVAILABLE,
+                })}"
+                .icon=${this._config.icon || stateIcon(stateObj)}
+                .disabled=${UNAVAILABLE_STATES.includes(stateObj.state)}
+                style=${styleMap({
+                  filter: this._computeSpeed(stateObj),
+                })}
+                @action=${this._handleAction}
+                .actionHandler=${actionHandler({
+                  hasHold: hasAction(this._config!.hold_action),
+                  hasDoubleClick: hasAction(this._config!.double_tap_action),
+                })}
+                tabindex="0"
+              ></ha-icon-button>
+            </div>
+          </div>
+
+          <div id="info">
+            ${UNAVAILABLE_STATES.includes(stateObj.state)
+              ? html`
+                  <div>
+                    ${computeStateDisplay(
+                      this.hass.localize,
+                      stateObj,
+                      this.hass.language
+                    )}
+                  </div>
+                `
+              : html` <div class="speed"></div> `}
+            ${this._config.name || computeStateName(stateObj)}
+          </div>
+        </div>
+      </ha-card>
+    `;
+  }
+
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    return hasConfigOrEntityChanged(this, changedProps);
+  }
+
+  protected updated(changedProps: PropertyValues): void {
+    super.updated(changedProps);
+    if (!this._config || !this.hass) {
+      return;
+    }
+
+    const stateObj = this.hass!.states[this._config!.entity];
+
+    if (!stateObj) {
+      return;
+    }
+
+    const oldHass = changedProps.get("hass") as HomeAssistant | undefined;
+    const oldConfig = changedProps.get("_config") as FanCardConfig | undefined;
+
+    if (
+      !oldHass ||
+      !oldConfig ||
+      oldHass.themes !== this.hass.themes ||
+      oldConfig.theme !== this._config.theme
+    ) {
+      applyThemesOnElement(this, this.hass.themes, this._config.theme);
+    }
+  }
+
+  private _dragEvent(e: any): void {
+    const stateObj = this.hass!.states[this._config!.entity] as FanEntity;
+    if (e.detail.value < 1) {
+      this.shadowRoot!.querySelector(".speed")!.innerHTML = `Off`;
```

Comment:
Please use the Query decorator and make this a private variable on the class.

```ts
@query(".speed") private _speedElement?: HTMLElement;
```

Reply:
And then use this everywhere
```suggestion
      this._speedElement.innerHTML = `Off`;
```